xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/internal/rss/DefaultNotificationRSSManager.java

```diff
@@ -0,0 +1,130 @@
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.internal.rss;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.localization.ContextualLocalizationManager;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.notifications.internal.ModelBridge;
+import org.xwiki.notifications.rss.NotificationRSSManager;
+import org.xwiki.notifications.rss.NotificationRSSRenderer;
+
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.feed.synd.SyndFeed;
+import com.rometools.rome.feed.synd.SyndFeedImpl;
+
+/**
+ * This is the default implementation of {@link NotificationRSSManager}.
+ *
+ * @version $Id$
+ * @since 9.6RC1
+ */
+@Component
+@Singleton
+public class DefaultNotificationRSSManager implements NotificationRSSManager
+{
+    @Inject
+    private ContextualLocalizationManager contextualLocalizationManager;
+
+    @Inject
+    private ModelBridge modelBridge;
+
+    @Inject
+    private DocumentReferenceResolver<String> documentReferenceResolver;
+
+    @Inject
+    private ComponentManager componentManager;
+
+    @Inject
+    private NotificationRSSRenderer defaultNotificationRSSRenderer;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public SyndFeed renderFeed(List<CompositeEvent> events)
+    {
+        SyndFeed feed = new SyndFeedImpl();
+
+        // Define the general properties of the rss
+        feed.setFeedType("rss_2.0");
+        feed.setTitle(this.contextualLocalizationManager.getTranslationPlain("notifications.rss.feedTitle"));
+
+        // Set the RSS feed link to the service generating the feed
+        feed.setLink(this.modelBridge.getDocumentURL(
+                this.documentReferenceResolver.resolve("XWiki.Notifications.Code.NotificationRSSService"),
```

Comment:
I don't understand why you parse a reference you know. You should put this reference in a static LocalDocumentReference instead.